ports/esp32/modwifi_csi.c

```diff
@@ -0,0 +1,705 @@
+    frame.smoothing = info->rx_ctrl.smoothing;
+    frame.not_sounding = info->rx_ctrl.not_sounding;
+    frame.aggregation = info->rx_ctrl.aggregation;
+    frame.stbc = info->rx_ctrl.stbc;
+    frame.fec_coding = info->rx_ctrl.fec_coding;
+    frame.sgi = info->rx_ctrl.sgi;
+    frame.ampdu_cnt = info->rx_ctrl.ampdu_cnt;
+    frame.secondary_channel = info->rx_ctrl.secondary_channel;
+    frame.ant = info->rx_ctrl.ant;
+    #endif
+
+    // Common fields available on all targets
+    frame.noise_floor = info->rx_ctrl.noise_floor;
+    frame.channel = info->rx_ctrl.channel;
+    frame.local_timestamp = info->rx_ctrl.timestamp;
+    frame.sig_len = info->rx_ctrl.sig_len;
+    frame.rx_state = info->rx_ctrl.rx_state;
+
+    // Copy MAC address
+    memcpy(frame.mac, info->mac, 6);
+
+    // Get timestamp
+    frame.timestamp_us = (uint32_t)esp_timer_get_time();
+
+    // Copy CSI data (info->buf is int8_t*, info->len is in bytes)
+    frame.len = info->len > CSI_MAX_DATA_LEN ? CSI_MAX_DATA_LEN : info->len;
+    if (info->buf && frame.len > 0) {
+        memcpy(frame.data, info->buf, frame.len);
+    }
+
+    // Write to circular buffer
+    csi_buffer_write(&g_csi_state.buffer, &frame);
+}
+
+// ============================================================================
+// CSI Control Functions
+// ============================================================================
+
+void wifi_csi_init(void) {
+    // Initialize with default configuration
+    if (!g_csi_state.buffer.initialized) {
+        csi_buffer_init(&g_csi_state.buffer, g_csi_state.config.buffer_size);
+    }
+}
+
+void wifi_csi_deinit(void) {
+    if (g_csi_state.enabled) {
+        wifi_csi_disable();
+    }
+    csi_buffer_deinit(&g_csi_state.buffer);
+}
+
+esp_err_t wifi_csi_enable(void) {
+    if (g_csi_state.enabled) {
+        ESP_LOGI(TAG, "Already enabled");
+        return ESP_OK;
+    }
+
+    esp_err_t ret;
+
+    ESP_LOGI(TAG, "Starting CSI enable sequence...");
+
+    // Ensure buffer is initialized
+    if (!g_csi_state.buffer.initialized) {
+        ESP_LOGI(TAG, "Initializing buffer (size=%lu)...",
+            (unsigned long)g_csi_state.config.buffer_size);
+        if (!csi_buffer_init(&g_csi_state.buffer, g_csi_state.config.buffer_size)) {
+            ESP_LOGE(TAG, "Failed to initialize buffer");
+            return ESP_ERR_NO_MEM;
+        }
+        ESP_LOGI(TAG, "Buffer initialized OK");
+    }
+
+    // Configure WiFi protocol mode
+    #if CONFIG_IDF_TARGET_ESP32C6 || CONFIG_IDF_TARGET_ESP32C5
+    // ESP32-C5/C6: Enable WiFi 6 (802.11ax) for improved performance and CSI
+    // capture
+    ESP_LOGI(TAG, "Setting WiFi protocol to 802.11b/g/n/ax...");
+    ret = esp_wifi_set_protocol(WIFI_IF_STA,
+        WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G |
+        WIFI_PROTOCOL_11N | WIFI_PROTOCOL_11AX);
+    ESP_LOGD(TAG, "esp_wifi_set_protocol returned: 0x%x", ret);
+    if (ret != ESP_OK) {
+        ESP_LOGW(TAG, "Failed to set WiFi protocol: 0x%x", ret);
```

Comment:
Is this meant to continue execution even if setting the protocol failed?  Same applies to the non-C5/C6 code fragment.

If execution has to stop on error, you can reduce the if/else block to just a single bitmask define and keep the rest of the code the same (the protocol set success log message loses its meaning then).